scylla/src/utils/pretty.rs

```diff
@@ -19,3 +23,310 @@ impl<'a> UpperHex for HexBytes<'a> {
+    use chrono::{NaiveDate, NaiveDateTime, NaiveTime};
+
+    use scylla_cql::frame::response::result::CqlValue;
+    use scylla_cql::frame::value::CqlDuration;
+
+    use crate::utils::pretty::CqlValueDisplayer;
+
+    #[test]
+    fn test_cql_value_displayer() {
+        assert_eq!(
+            format!("{}", CqlValueDisplayer(CqlValue::Boolean(true))),
+            "true"
+        );
+        assert_eq!(format!("{}", CqlValueDisplayer(CqlValue::Int(123))), "123");
+        assert_eq!(
+            format!(
+                "{}",
+                CqlValueDisplayer(CqlValue::Decimal(BigDecimal::from_str("123.456").unwrap()))
+            ),
+            "123.456"
+        );
+        assert_eq!(
+            format!("{}", CqlValueDisplayer(CqlValue::Float(12.75))),
+            "12.75"
+        );
+        assert_eq!(
+            format!(
+                "{}",
+                CqlValueDisplayer(CqlValue::Text("Ala ma kota".to_owned()))
+            ),
+            "'Ala ma kota'"
+        );
+        assert_eq!(
+            format!("{}", CqlValueDisplayer(CqlValue::Text("Foo's".to_owned()))),
+            "'Foo''s'"
+        );
+
+        // Time types are the most tricky
+        assert_eq!(
+            format!("{}", CqlValueDisplayer(CqlValue::Date(40 + (1 << 31)))),
+            "'1970-02-10'"
+        );
+        assert_eq!(
+            format!(
+                "{}",
+                CqlValueDisplayer(CqlValue::Duration(CqlDuration {
+                    months: 1,
+                    days: 2,
+                    nanoseconds: 3,
+                }))
+            ),
+            "1mo2d3ns"
+        );
+        let t = Duration::from_nanos(123)
+            + Duration::from_secs(4)
+            + Duration::from_secs(5 * 60)
+            + Duration::from_secs(6 * 60 * 60);
+        assert_eq!(
+            format!(
+                "{}",
+                CqlValueDisplayer(CqlValue::Time(chrono::Duration::from_std(t).unwrap()))
+            ),
+            "'06:05:04.000000123'"
+        );
+
+        let t = NaiveDate::from_ymd_opt(2005, 4, 2)
+            .unwrap()
+            .and_time(NaiveTime::from_hms_opt(19, 37, 42).unwrap());
+        assert_eq!(
+            format!(
+                "{}",
+                CqlValueDisplayer(CqlValue::Timestamp(
+                    t.signed_duration_since(NaiveDateTime::default())
+                ))
+            ),
+            "'2005-04-02 19:37:42.000+0000'"
+        );
+
+        // Compound types
+        let list_or_set = vec![CqlValue::Int(1), CqlValue::Int(3), CqlValue::Int(2)];
+        assert_eq!(
+            format!("{}", CqlValueDisplayer(CqlValue::List(list_or_set.clone()))),
+            "[1,3,2]"
+        );
+        assert_eq!(
+            format!("{}", CqlValueDisplayer(CqlValue::Set(list_or_set.clone()))),
+            "{1,3,2}"
+        );
+
+        let mut tuple: Vec<_> = list_or_set.into_iter().map(Some).collect();
+        tuple.push(None);
```

Comment:
Possibly cleaner:

Reply:
```suggestion
        let tuple = Vec::from_iter(list_or_set.into_iter().map(Some).chain(std::iter::once(None)));
```

Reply:
`FromIterator` should be encouraged, as it reduces verbosity induced by `collect()` (either type hint or turbofish).

Reply:
> `FromIterator` should be encouraged, as it reduces verbosity induced by `collect()` (either type hint or turbofish).

Why? I quite like `.collect()` to be fair. It leads to more natural flow in the code. When you work with iterators, you perform a series of steps (such as map, filter, etc.) and then finally you `.collect()` the results.

Reply:
This is a valid counterargument.